sans/reduction.ipynb

```diff
@@ -210,6 +210,55 @@
     "    return (pixel_size * pixel_length) / (L2 * L2)"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def q_resolution(lam_edges, moderator, d):\n",
+    "    moderator = sc.rebin(moderator, 'wavelength', lam_edges)\n",
+    "    \n",
+    "    d_lam = lam_edges['wavelength', 1:] - lam_edges['wavelength', :-1] # bin widths\n",
+    "    lam = 0.5 * (lam_edges['wavelength', 1:] + lam_edges['wavelength', :-1]) # bin centres\n",
+    "    \n",
+    "    r2 = sc.Variable(value=4.0824829046386295/1000, unit=sc.units.m) # sample aperture radius\n",
+    "    r1 = sc.Variable(value=14.433756729740645/1000, unit=sc.units.m) # source aperture radius\n",
+    "    dr = sc.Variable(value=8.0/1000, unit=sc.units.m) # virtual ring width on detector\n",
+    "    \n",
+    "    l2 = sc.neutron.l2(d)\n",
+    "    l1 = sc.Variable(value=5.0, unit=sc.units.m) # should be sc.neutron.l1(d) = 25.6m ?\n",
+    "    theta = sc.neutron.scattering_angle(d)  \n",
+    "\n",
+    "    inv_l3 = (l1 + l2) / (l1*l2)\n",
+    "    \n",
+    "    # Terms in Mildner and Carpenter equation. See https://docs.mantidproject.org/nightly/algorithms/TOFSANSResolutionByPixel-v1.html\n",
+    "    a1 = (r1/l1)*(r1/l1) * 3.0\n",
+    "    a2 = (r2*inv_l3)*(r2*inv_l3) * 3.0\n",
+    "    a3 = (dr/l2) * (dr/l2) \n",
+    "    q = 4.0 * np.pi * sc.sin(theta) * sc.reciprocal(lam) # keep in wav dim to prevent broadcast\n",
+    "    \n",
+    "    h = 6.62607004E-34\n",
+    "    m = 1.6749E-27\n",
+    "    k = sc.Variable(value=h/m, unit=(sc.units.m * sc.units.m)/sc.units.s) \n",
+    "    \n",
+    "    moderator.data.values *= 1e-6 # us to seconds\n",
+    "    moderator.data.unit = sc.units.s # fixup moderator units to microseconds. Previously dimensionless\n",
+    "    \n",
+    "    # Calcualte moderator spread\n",
+    "    s = moderator.data * sc.reciprocal(l1 + l2) # times to inv velocity\n",
+    "    s *= k # inv velocity to wavelength\n",
+    "    s.unit = sc.units.angstrom\n",
+    "    s *= 1e10\n",
```

Comment:
Interesting! Looking in some more detail at the moderator spread calculation shows that this is simply a conversion from TOF to wavelength, except that the time is in the values of `moderator`, not in the coord. Please verify, but if true we can do something like

```python
mod = sc.Dataset(coords={'tof':moderator.data, 'source_position':moderator.attrs['source_position'], ...)
s = sc.neutron.convert(mod, 'tof', 'wavelength').coords['wavelength']
```
Avoiding the need for duplicating physical constants and unit conversions.

Interestingly, this hints at a wrong abstraction for unit conversions: Maybe unit conversion should be exposed in a different way, such that it cannot just convert coords. Will need to think about ths.

Reply:
Indeed, would be nice if this was not limited to coordinates.

Reply:
@OwenArnold Try if this gives the same result:

```python
tof = moderator.data.copy()
tof.variances = None # shortcoming of Mantid or Mantid converter?
tof.rename_dims({'wavelength':'tof'}) # TODO overly restrictive check in convert (rename)
mod = sc.Dataset(coords={
    'tof':tof,
    'position':sample.coords['position'],
    'source_position':sample.coords['source_position'],
    'sample_position':sample.coords['sample_position']})
sc.neutron.convert(mod, 'tof', 'wavelength').coords['wavelength']
```